src/app/shared/courses/courses.service.ts

```diff
@@ -51,30 +58,50 @@ export class CoursesService {
             authors: null,
             isTopRated
         };
-        this.http.post('http://localhost:3004/courses/', course, httpOptions);
+        this.http.post(ROOT_URL, course, httpOptions)
+            .subscribe()
+            .add(() => { _this.preloadingService.setLoginStatus(false); });
         return course;
     }
 
-    update(id: number, name: string, length: number, description: string, isTopRated: boolean): Course {
-        const course: Course = this.getItemById(id);
-        course.name = name;
-        course.length = length;
-        course.description = description;
-        course.isTopRated = isTopRated;
-        return course;
+    update(id: number, name: string, length: number, description: string, isTopRated: boolean): void {
+        const _this = this;
+        this.getItemById(id)
+            .subscribe(
+                course => {
+                    console.log(course);
+                    course.name = name;
+                    course.length = length;
+                    course.description = description;
+                    course.isTopRated = isTopRated;
+                    
+                    _this.http.patch(`${ROOT_URL}${id}`, course, httpOptions)
+                        .subscribe()
+                        .add(() => { _this.preloadingService.setLoginStatus(false); });
+                },
+                error => { console.log(error) }
+            );
     }
 
     removeCourse(id: number): Course[] {
         return this.courses = this.courses.filter(course => course.id !== id);
     }
 
     removeServerCourse(id: number): void {
-        this.http.delete(`http://localhost:3004/courses/${id}`);
+        const _this = this;
+        this.http.delete(`${ROOT_URL}${id}`)
+        .subscribe()
+        .add(() => { _this.preloadingService.setLoginStatus(false); });
     }
```

Comment:
Your service might return pure streams, especially when angular forces to use it everywhere with `async` pipe
```typescript
removeById(id) {
  return this.http.delete(`${ROOT_URL}${id}`).pipe(
  tap(() => this.preloadingService.setLoginStatus(false)),
);
}
```
I would suggest using an Interceptor for showing/hiding loader before/after a request